src/main/java/org/jabref/gui/externalfiles/FindFullTextAction.java

```diff
@@ -39,6 +43,10 @@
     private final BasePanel basePanel;
     private final DialogService dialogService;
 
+    //Stuff for downloading full texts
+    private final FulltextFetchers fetcher = new FulltextFetchers(JabRefPreferences.getInstance().getImportFormatPreferences());
```

Comment:
We try to get rid of explicit calling JabRefPreferences., the ideal solution would be pass the ImportFormatPrefernces as parameter in the constructor.

Reply:
And while you are onto this class, you could refactor it to extend the SimpleCommand class (see for example WriteXMP Action)

Reply:
@Siedlerchr So, it should not implement the BaseAction interface anymore then, correct?

Reply:
Yes, replace the base action with SimpleCommand, instead of an action method you have an execute method, it's technically just a different name, the behavior is the same and you can just rename it

Reply:
@Siedlerchr BasePanel.java doesn't like that because now the second argument in the constructor is no longer a "BaseAction".  How should I proceed?

Reply:
you can get rid of that action entry in the action map, look at the setup of the actions in the menus,

Reply:
Thanks for the help! I think I've got everything fixed up.